integration: migrate TestAPICreateDeletePredefinedNetworks from integration-cli

5h4rk-lab opened the pull request:
- supersedes, closes https://github.com/moby/moby/pull/51268
- addresses https://github.com/moby/moby/issues/50159


- What I did
Migrated the legacy test TestAPICreateDeletePredefinedNetworks from integration-cli/docker_api_network_test.go to the new integration suite in integration/network/create_delete_predefined_test.go.

- How I did it
Reimplemented the test using the Go client APIs (client.NetworkCreate, client.NetworkRemove, client.NetworkList) instead of the old CLI test framework.
The test asserts that creating or deleting the predefined networks (bridge, host, none) fails, and verifies that each still exists afterward.

- How to verify it
Run the test with:

```
DOCKER_API_VERSION=1.51 \
TESTDIRS='./integration/network' \
TESTFLAGS='-test.run TestCreateDeletePredefinedNetworks' \
make test-integration
```

- Human readable description for the release notes

Migrate predefined network test from integration-cli to the integration test suite.  
This ensures that attempts to create or delete built-in networks (`bridge`, `host`, `none`) fail as expected, and validates they remain available.

![Cute puppy](https://place-puppy.com/puppy/y:400/x:300)

5h4rk-lab committed:
integration: migrate TestAPICreateDeletePredefinedNetworks from integration-cli

Signed-off-by: Go Charan Kilaru <sharkmagic07@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah reviewed (changes requested):
Left comments, but was also writing along, so if you want to copy / paste my suggestions; here's the whole file;


```go
package network

import (
	"slices"
	"testing"

	cerrdefs "github.com/containerd/errdefs"
	"github.com/moby/moby/client"
	"gotest.tools/v3/assert"
	is "gotest.tools/v3/assert/cmp"
)

func TestCreateDeletePredefinedNetworks(t *testing.T) {
	ctx := setupTest(t)

	predefinedNetworks := []string{"bridge", "host", "none"}
	if testEnv.DaemonInfo.OSType == "windows" {
		predefinedNetworks = []string{"nat", "none"}
	}

	apiClient := testEnv.APIClient()

	nws, err := apiClient.NetworkList(ctx, client.NetworkListOptions{})
	assert.NilError(t, err)

	var actual []string
	for _, nw := range nws {
		if slices.Contains(predefinedNetworks, nw.Name) {
			actual = append(actual, nw.Name)
		}
	}
	slices.Sort(actual)
	assert.Check(t, is.DeepEqual(actual, predefinedNetworks))

	for _, name := range predefinedNetworks {
		t.Run(name, func(t *testing.T) {
			// Creating a predefined network must fail.
			_, err := apiClient.NetworkCreate(ctx, name, client.NetworkCreateOptions{})
			assert.Check(t, is.ErrorContains(err, "operation is not permitted on predefined"))
			assert.Check(t, is.ErrorType(err, cerrdefs.IsPermissionDenied))

			// Deleting a predefined network must fail.
			err = apiClient.NetworkRemove(ctx, name)
			assert.Check(t, is.ErrorContains(err, "is a pre-defined network and cannot be removed"))
			assert.Check(t, is.ErrorType(err, cerrdefs.IsPermissionDenied))

			// Sanity: predefined network should still exist.
			_, err = apiClient.NetworkInspect(ctx, name, client.NetworkInspectOptions{})
			assert.NilError(t, err)
		})
	}
}
```

5h4rk-lab commented:
Updated: removed legacy TestDockerNetworkLsDefault from integration-cli since the new integration test now verifies default/predefined networks exist (via NetworkList + comparison). Re-ran the scoped integration test; still green.

thaJeztah reviewed (approved):
LGTM if CI is happy


did a quick rebase, and moved the test to the file (as mentioned above)

thaJeztah commented:
CI failures are unrelated (flaky tests and GitHub actions runners flakiness)
